src/mon/Monitor.cc

```diff
@@ -986,6 +986,16 @@ void Monitor::shutdown()
 
   remove_all_sessions();
 
+  log_client.shutdown();
+
+  // unlock before msgr shutdown...
+  lock.Unlock();
+
+  messenger->shutdown();  // last thing!  ceph_mon.cc will delete mon.
```

Comment:
the comment is not correct anymore. probably we could revise it like

> shutdown messenger before removing logger from perfcounter collection, otherwise _ms_dispatch() will try to update deleted logger

Reply:
All right.I will change